crypto/yubihsm.go

```diff
@@ -258,7 +260,26 @@ func (privKey *YubiHsmPrivateKey) ExportPubKey() error {
 	return err
 }
 
-func (privKey *YubiHsmPrivateKey) yubiHsmSecp256k1Sign(hash []byte) (sig []byte, err error) {
+func (privKey *YubiHsmPrivateKey) getSigRecID(hash []byte, sig []byte) (byte, error) {
+	recIds := []byte{0x00, 0x01, 0x02, 0x03}
+
+	for i := 0; i < len(recIds); i++ {
+		tmpSig := make([]byte, 65)
+		copy(tmpSig[:], sig)
+		tmpSig[64] = recIds[i]
+		pubKeyBytes, err := secp256k1.RecoverPubkey(hash, tmpSig[:])
+		if err == nil && bytes.Equal(pubKeyBytes, privKey.pubKeyUncompressed) {
+			return recIds[i], nil
+		}
+	}
+
+	return 0x04, fmt.Errorf("Unable to get recovery public key from signature")
```

Comment:
There has to be a better way to figure out the recovery ID...
Does this make sense https://ethereum.stackexchange.com/questions/42455/during-ecdsa-signing-how-do-i-generate-the-recovery-id ?
And this talks about the same approach https://www.reddit.com/r/crypto/comments/6gcypt/how_do_you_determine_a_recovery_id_for_ecdsa/